specs/unified_kernel_image.md

```diff
@@ -40,20 +40,39 @@ that glues the listed components together, generating one PE binary that then ca
+* The Linux kernel in the `.linux` PE section.
 * Optionally, information describing the OS this kernel is intended for, in the `.osrel` section. The contents of this section are derived from `/etc/os-release` of the target OS. They can be useful for presentation of the UKI in the boot loader menu, and ordering it against other entries using the included version information.
-* Optionally, information describing kernel release information (i.e. `uname -r` output) in the `.uname` section. This is also useful for presentation of the UKI in the boot loader menu, and ordering it against other entries.
+* Optionally, the kernel command line in the `.cmdline` section. If this is absent, the loader implementation may allow local overrides instead.
+* The initrd that the kernel shall unpack and invoke, in the `.initrd` section.
 * Optionally, a splash image to bring to screen before transitioning into the Linux kernel, in the `.splash` section.
 * Optionally, one or more compiled Device Trees, for systems which need it, each in its separate `.dtb` section. If multiple `.dtb` sections exist then one of them is selected according to an implementation-specific algorithm.
-* Optionally, the public part of a public-private key pair in PEM format used to sign the image, in the `.pcrpkey` section.
-* Optionally, a JSON file encoding expected PCR 11 hash values seen from userspace once the UKI has booted up, along with signatures of these expected PCR 11 hash values, in the `.pcrsig` section. The signatures must also match the abovementioned key pair.
+* Optionally, information describing kernel release information (i.e. `uname -r` output) in the `.uname` section. This is also useful for presentation of the UKI in the boot loader menu, and ordering it against other entries.
 * Optionally, a CSV file encoding the SBAT metadata for the image, in the `.sbat` section. The [SBAT format is defined by the Shim project](https://github.com/rhboot/shim/blob/main/SBAT.md), and used for UEFI revocation purposes.
+* Optionally, a JSON file encoding expected PCR 11 hash values seen from userspace once the UKI has booted up, along with signatures of these expected PCR 11 hash values, in the `.pcrsig` section. The signatures must also match the key pair described below.
+* Optionally, the public part of a public-private key pair in PEM format used to sign the image, in the `.pcrpkey` section.
+
+Note that all of the sections defined above are singletons: they may
+appear once at most – except for the `.dtb` section which may be
+appear more than once.
+
+On systems with a TPM the UEFI boot stub shall measure the payload
+sections (i.e. the PE sections listed above, starting from the
+`.linux` one), in the order as listed above (which should be
+considered the *canonical* *order*). Specifically, for each section two
+measurements shall be made into PCR 11 with event code EV_IPL:
+
+1. First, the section name in ASCII (including a trailing NUL byte)
```

Comment:
We already have numbering, so drop "First, ", "Second, ".

The section name in ASCII may contain multiple NUL bytes to align to the padding… Maybe clarify that exactly one should be measured? (Or whatever the rule is.)